ldelf/ta_elf.c

```diff
@@ -805,18 +806,22 @@ static void add_deps_from_segment(struct ta_elf *elf, unsigned int type,
 	assert(!(memsz % dyn_entsize));
 	num_dyns = memsz / dyn_entsize;
 
-	for (n = 0; n < num_dyns; n++) {
+	for (n = 0; n < num_dyns && !(str_tab && str_tab_sz); n++) {
```

Comment:
nitpicking. for consistency of the `for ()` loop, I suggest this change:
```C
	for (n = 0, str_tab = NULL, str_tab_sz = 0;
	     n < num_dyns && !(str_tab && str_tab_sz); n++) {

Reply:
That is a bit excessive and doesn't make it easier to read in my eyes.
As for consistency, that pattern isn't clear to me.